### PR TITLE
Bootstrap Container Version v0.1.0

### PR DESCRIPTION
**Pull request by CSHieuV**
**Description of Changes:** 

Code to create Bootstrap container to run on a Bottlerocket host.

**Testing Done:**

- A user can create the Bottlerocket instance and validate that changes specified in the user script have been applied successfully

Some user data used for testing can be seen below:

```
#!/usr/bin/env sh
set -euo pipefail
# Create the directory
mkdir -p /var/lib/my_directory
# Set API client configurations
apiclient set --json '{"settings": {"oci-defaults": {"resource-limits": {"max-open-files": {"soft-limit": 4294967296, "hard-limit": 8589934592}}}}}'
# Load kernel module
if command -v modprobe > /dev/null 2>&1; then
  modprobe dummy
fi
echo "User-data script executed."
```

Output from testing:
```
Test passed: Finished bootstrap container
Running Apiclient Setting Test...
API client output: {
  "settings": {
    "oci-defaults": {
      "resource-limits": {
        "max-open-files": {
          "hard-limit": 8589934592,
          "soft-limit": 4294967296
        }
      }
    }
  }
}
Apiclient Setting Test: PASSED
Running Kernel Test...
Kernel Test: PASSED
Running Directory Creation Test...
Directory Creation Test: PASSED
```

- Testing for CA certs and s3 buckets also passed

**Related PR's**

Closes issue [#2010](https://github.com/bottlerocket-os/bottlerocket/issues/2010)

**Terms of Contribution**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    command -v mdadm
+
+# Copy the wrapper script into the container
+COPY bootstrap-script.sh /usr/local/bin/bootstrap-script.sh
+
+# Make the wrapper script executable
+RUN chmod +x /usr/local/bin/bootstrap-script.sh
+
+# Set the wrapper script as the entry point
+ENTRYPOINT ["/usr/local/bin/bootstrap-script.sh"]
```
Comment (yeazelm) [inline Dockerfile]: nit: looks like we should include a newline here.

File: README.md [hunk 2]
```diff
@@ -1,17 +1,63 @@
...
-Be sure to:
+## Using the Container Image
 
-* Change the title in this README
-* Edit your repository description on GitHub
+For more information on settings, please refer to the [Bottlerocket Documentation](https://bottlerocket.dev/) in order to configure your files. 
+To use the container image, create a toml file with the following configurations. 
 
-## Security
+### Sample toml configuration
```
Comment (yeazelm) [inline README.md]: nit:
```suggestion
### Sample TOML configuration
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,63 @@
...
+mode="once"
+user-data="user-base64-encoded-bootstrap-script"
+```
 
-## License
+Afterwards, create the Bottlerocket instance using the toml file. 
+The settings that you changed within the user's bootstrap script should be seen within the bottlerocket instance. 
 
-This project is licensed under the Apache-2.0 License.
+### Example runthrough
```
Comment (yeazelm) [inline README.md]: another nit:
```suggestion
### Example Walkthrough
```

**Review by yeazelm — APPROVED** [review]
LGTM!

**Review by vyaghras — APPROVED** [review]
:shipit:

**Pull request merged**